### PR TITLE
Support for {site} token for project tasks

### LINKED ISSUE #646: Include {site} Token in Oppgaveliste
**Is your feature request related to a problem? Please describe.**
It would be usefull to connect a 'oppgave' (Planner task) with a document within the project.
This can be done manually directly on the Task as of today.
There is no support for adding this link in the source master list 'Planneroppgaver' on the Hub.
And have this automtically populated by the 'Prosjektkonfigurasjon' wizard.

**Describe the solution you'd like**
Add support for Tokens in the Planneroppgvaer list on the hub.
Specialy the Token {site} would be helpful. By using this token the 'Prosjektkonfigurasjon' wizard can translate and replace with the new projects actual site-url.
Together with preselected document content the Planner task should be able to provide a active link to a corresponding document.

**Additional context**

### PR DESCRIPTION
**Pull request by TinHoangVu**
### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [x] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [x] Check the commit's or even all commits' message 
- [x] Check if your code additions will fail linting checks
- [x] Remember: Add issue description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the **ID of the Issue** associated with this PR
- [x] Documentation: Have a look at the [PP365 User manual](https://puzzlepart.github.io/prosjektportalen-manual/) and consider the need for updates to be made. Updates can be done directly into the 'Kladd' branch or by providing information to test team for implementation.

### Description

Support for {site} token for project tasks.

### How to test

- [x] 1. Add document url to plannertask list on hub. Eks: {site}/Delte%20dokumenter%20/mandat%20for%20konseptfasen.docx?web=1;Mandat for konseptfasen;Word
- [x] 2. Create a new project and include plannertasks.
- [x] 3. See that the attachment is included in the tasks.

### Relevant issues (if applicable)
Closes #646

💔Thank you!

### CODE REVIEW

**Comment by olemp** [comment]
@TinHoangVu This should work now.

```text
{site}/Delte%20dokumenter/Mandat for konseptfasen.docx?web=1;Mandat for konseptfasen;Word
```

**Comment by olemp** [comment]
@Petterandre Was this the issue you tried to fix [here](https://github.com/Puzzlepart/prosjektportalen365/blame/main/SharePointFramework/ProjectExtensions/src/projectSetup/index.ts#L394)?

image

Not sure how the map without return-statement fixed this though.. Can we fix this in a cleaner way.. somehow?

FYI @TinHoangVu

**Comment by olemp** [comment]
@TinHoangVu It should work after the latests commits. Finally understood the code..

**Review by olemp — APPROVED** [review]

File: SharePointFramework/ProjectExtensions/src/projectSetup/index.ts [hunk 1]
```diff
@@ -331,62 +331,43 @@ export default class ProjectSetup extends BaseApplicationCustomizer<IProjectSetu
         data: {},
         level: LogLevel.Info
       })
-
```
Comment (Remi749) [inline SharePointFramework/ProjectExtensions/src/projectSetup/index.ts:335]: oof

**Pull request merged**